### PR TITLE
Hdrp/reflection probe scale perf

### PR DESCRIPTION
**Pull request by fredericv-unity3d**
### Purpose of this PR
Improved the registration and the unregistration of reflection probes.

(For a 10 000 probe test, the registration time take 1 000 ms before and 8 ms after).

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Baked probe on a project
- [x] Used planar reflection on a project and verified their culling

**Automated Tests**:
- Added a performance test for measurements of the reflection probe registration.

---

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/HDBakedReflectionSystem.cs [hunk 1]
```diff
@@ -326,9 +327,8 @@ public static bool BakeProbes(IList<HDProbe> bakedProbes)
             var cubeRT = HDRenderUtilities.CreateReflectionProbeRenderTarget(cubemapSize);
 
             // Render and write the result to disk
-            for (int i = 0; i < bakedProbes.Count; ++i)
+            foreach (var probe in bakedProbes)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/HDBakedReflectionSystem.cs:330]: iirc foreach cause GC.Alloc no ?

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/HDBakedReflectionSystem.cs:330]: IIRC Antoine have remove all of them to be able to have 0 GC.Alloc

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/HDBakedReflectionSystem.cs:330]: Most of the foreach are fine but some of them allocate memory due to the kind of iterator there are using. 
Whether this allocates memory or not will depend on the type of `bakedProbes`

Comment (fredericv-unity3d) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/HDBakedReflectionSystem.cs:330]: They are reference type, no value type. So there is no boxing during the foreach loop

**Comment by sebastienlagarde** [comment]
Yamato is failing, pleae fix: https://yamato-artifactviewer.prd.cds.internal.unity3d.com/86056f9d-e6b2-4952-83e3-149f196f63fc%2Flogs%2FTestProjects%2FHDRP_Tests%2Fupm-ci~%2Ftest-results/TestReportV1.html

**Pull request merged**